### PR TITLE
Update the proto, go, and c++ packages

### PR DESCRIPTION
**Pull request by Alfus**
To match the following:
- proto: cel.expr.* / "cel/expr/*"
- c++: cel::expr::* / "cel/expr/*"
- go: cel.dev/expr
- java: dev.cel.expr (unchanged)

The tld is typically not included in proto/c++, and was in the wrong spot for the go package.

This leaves the existing protos inplace, to allow for a transition to the new packages.

### CODE REVIEW

**Comment by jcking** [comment]
Not sure I am super excited about the C++ namespace `cel::expr`. It's a bit redundant, almost like saying `ATM machine`.

Maybe we are better off adding a suffix to all the top-level messages/enums, such as `Proto`? That way it is guaranteed to not collide with native names.

**Comment by Alfus** [comment]
cel::expr is the 'expression' part of the C(E)L. There should also be a parser and an evaluator for the full language, for example.

**Comment by Alfus** [comment]
A rename of the messages would be a **much** larger transition for the open source libraries that have already adopted these protos.

**Comment by Alfus** [comment]
(also, pretty sure you need at least one sub package in go and maybe python)

File: proto/cel/expr/checked.proto [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package cel.expr;
```
Comment (l46kok) [inline proto/cel/expr/checked.proto:17]: Why are we not including the tld in the main proto package name?

Comment (Alfus) [reply proto/cel/expr/checked.proto:17]: The real requirement is: The proto package should match both the c++ namespace and the folders.

This is the (defacto?) standard way to reconcile that c++, python, es, etc, which do not typically include the tld in namespaces/packages. It is only go (which uses it as real dns address) and java (which shoves the rdomain in the namespace per the style guide) that use the tld.

**Comment by TristonianJones** [comment]
@Alfus can you sync with HEAD and push again? I've updated the config for GCB, but I don't think it'll take in this PR until its synced.

**Comment by Alfus** [comment]
I could also delete some of the old definitions. It seems to just be a lot to compute

**Review by TristonianJones — APPROVED** [review]

**Pull request merged**